packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -80,5 +82,9 @@ export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLE
         disconnectedCallback() {
             disconnectRootElement(this);
         }
+        attributeChangedCallback(name: string, oldValue: any, newValue: any) {
+            attributeChangedCallback.call(this, name, oldValue, newValue);
+        }
+        static observedAttributes = observedAttributes;
```

Comment:
this is one of the parts that worries me a little bit!

Reply:
@caridy For the sake of my fuzzy memory, what part of it worries you?

Reply:
I need to think more about it... but the fact that we now rely on observed attributes in the class that we produce, might introduce other issues with the inheritance path outside of LWC, and probably also conflict with authors list of attributes. I don't remember how `observedAttributes` from this record is constructed.

Reply:
If I'm not mistaken, we handle inheritance of observed attributes in `base-bridge-element.ts`:

https://github.com/salesforce/lwc/blob/838035de35f670bfedd9772ad78b573a78586d58/packages/%40lwc/engine-core/src/framework/base-bridge-element.ts#L187-L193

Reply:
right, but beyond that, I don't know.

Reply:
yeah, I don't doubt that, but why was this change needed in the first place? If the custom element inherit the right thing... then you should be able to get the observedAttributes and others from there rather than rebuilding that yourself.